### ISSUE TITLE
[REFACTORING] Evaluate the impact of removing the client side compression for backup blocks

### ISSUE THREAD

**Issue by joshimoo** [longhorn/longhorn#1409]
- This would reduce the cpu load on the nodes when creating a backup.
- This would increase the backup size, by how much?
- This would allow download and check operations to make sure that the backups crc still matches with the backup contents, currently we would need to download uncompress, crc check the data.

While doing this think about some alternatives, maybe keep a crc/hash of the compressed data, etc.

**Comment by joshimoo** [comment]
@derekbit thanks for the evaluation, can you summarize the testing procedure that you went through:

My assumption is that you created 2 longhorn volumes. volume1 you filled with uncompressable data (/dev/urandom) while volume2 was filled with compressable data. Then always did clean backups (inital backup) via longhorn?

- How did you measure the compression ratio/times?
- Did you implement the other compression algorithms into the backupstore as a test for evaluation?

In general your results look quite nice, I left a comment with additional implementation notes in https://github.com/longhorn/longhorn/issues/3175

Paralizing the backup process is a good idea, but has some constraints/concerns: 
- we need to implement rate limiting support for s3 #2810 
- we need to consider resource consumption/limits (io / cpu / memory)
- we don't want to starve the io/cpu of the nodes, since backup priority should be lower then active io/cpu from volumes in use.

**Comment by iosifnicolae2** [comment]
Looking forward for having `concurrent files transfer workers increasing the transfer speed` or maybe to be able to increase the backup/restore block size to a higher value so we can achieve higher throughput.

I appreciate a lot your effort on making Longhorn the best storage solution for on-premise Kubernetes clusters.

**Comment by derekbit** [comment]
Closed. Track in https://github.com/longhorn/longhorn/issues/5189.
